syntax = "proto3";
package pt.ulisboa.tecnico.NameServer.contract;

message RegisterRequest {
    string service_name = 1;
    string server_qualifier = 2;
    string server_address = 3;
}

message RegisterResponse {}

message DeleteRequest{
    string service_name = 1;
    string server_address = 2;
}

message DeleteResponse{}

message LookupRequest{
    string service_name = 1;
    string server_qualifier = 2;
}

message LookupResponse{
    repeated string servers = 1;
}
service NameService {
    rpc register(RegisterRequest) returns (RegisterResponse);
    rpc lookup(LookupRequest) returns (LookupResponse);
    rpc delete(DeleteRequest) returns (DeleteResponse);
}
